jetson/nav/search/searchStateMachine.cpp

```diff
@@ -246,13 +247,38 @@ NavState SearchStateMachine::executeDriveToTarget( Rover* phoebe, const rapidjso
         roverStateMachine->updateObstacleDistance( phoebe->roverStatus().obstacle().distance );
         return NavState::SearchTurnAroundObs;
     }
+    
+    DriveStatus driveStatus;
 
-    DriveStatus driveStatus = phoebe->drive( phoebe->roverStatus().target().distance,
+    if( phoebe->roverStatus().target2().distance > 0 )
+    {
+        if( phoebe->roverStatus().target().distance < phoebe->roverStatus().target2().distance ) 
```

Comment:
This seems quite a bit more complicated than before. Let's add a high level comment before this block of code to describe what should be happening.